cylc/flow/suite_files.py

```diff
@@ -611,6 +663,74 @@ def clean(reg):
     _remove_empty_reg_parents(reg, run_dir)
 
 
+def remote_clean(reg, platform_names):
+    """Run subprocesses to clean workflows on remote install targets
+    (skip localhost), given a set of platform names to look up.
+
+    Args:
+        reg (str): Workflow name.
+        platform_names (list): List of platform names to look up in the global
+            config, in order to determine the install targets to clean on.
+    """
+    try:
+        install_targets_map = (
+            get_install_target_to_platforms_map(platform_names))
+    except PlatformLookupError as exc:
+        raise PlatformLookupError(
+            "Cannot clean on remote platforms as the workflow database is "
+            f"out of date/inconsistent with the global config - {exc}")
+
+    queue = []
+    for target, platforms in install_targets_map.items():
+        if target == 'localhost':
+            continue
+        shuffle(platforms)
+        # Issue ssh command:
+        queue.append(
+            (_remote_clean_cmd(reg, platforms[0]), target, platforms)
+        )
+    for proc, target, platforms in queue:
+        ret_code = proc.wait()
+        out, err = (f.decode() for f in proc.communicate())
+        if out:
+            LOG.info(out)
+        if err:
+            LOG.warning(err)
+        if ret_code:
+            # Try again on the next platform for this install target:
+            this_platform = platforms.pop(0)
+            exc = TaskRemoteMgmtError(
+                TaskRemoteMgmtError.MSG_TIDY, this_platform['name'],
+                " ".join(proc.args), ret_code, out, err)
+            LOG.error(exc)
+            if platforms:
+                queue.append(
+                    (_remote_clean_cmd(reg, platforms[0]), target, platforms)
+                )
+            else:  # Exhausted list of platforms
+                raise CylcError(
+                    f"Could not clean on install target: {target}")
+
+
+def _remote_clean_cmd(reg, platform):
+    """Remove a stopped workflow on a remote host.
+
+    Call "cylc remote-clean" over ssh and return the subprocess.
+
+    Args:
+        reg (str): Workflow name.
+        platform (dict): Config for the platform on which to remove the
+            workflow.
+    """
+    LOG.info(
+        f'Cleaning on install target: {platform["install target"]} '
+        f'(platform: {platform["name"]})')
+    cmd = ['remote-clean', reg, get_remote_suite_run_dir(platform, reg)]
```

Comment:
Can just call plain `cylc clean` to avoid duplicating it as `remote-clean`. The `remote_*` scripts are things that are called automatically by Cylc during the course of running a flow.

Reply:
At the moment, the difference between `clean` and `remote-clean` is the former includes looking up the platforms in the database, and the latter includes an arg for specifying the cylc-run dir on the machine, which shouldn't be an arg for the local command

Reply:
Ok, I've replaced `remote-clean` with an internal-use option for normal `clean`, so it will be `cylc clean --local-only-run-dir=<run_dir>`. Will push later today